src/nodesessionmetadata.cpp

```diff
@@ -0,0 +1,106 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "nodesessionmetadata.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+#include "logging.hpp"
+
+namespace ovms {
+
+std::vector<NodeSessionMetadata> NodeSessionMetadata::generateSubsessions(const std::string& nodeName, session_id_t subsessionSize) {
+    if (nodeName.size() == 0) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Tried to generate subsession with empty node name");
+        throw std::logic_error("Cannot generate session with empty parent name");
```

Comment:
Will throwing an exception print its message to console? If so, we do not need additional logging in line 28.

Reply:
It will print the error on the console but not into the potential log files that user may specify, hence additional log line.